### PR TITLE
fix duplicate ips

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Without this, i have seen `2020-03-16T02:48:08.352325Z	info	Proxy role: &model.Proxy{ClusterID:"", Type:"sidecar", IPAddresses:[]string{"10.244.0.53", "10.244.0.53", "fe80::78c4:bff:fe2b:8378"}, ID:"productpage-v1-85b9bf9cd7-trfvx.istio-bookinfo-1-40038", Locality:(*envoy_api_v2_core.Locality)(nil), DNSDomain:"istio-bookinfo-1-40038.svc.cluster.local", ConfigNamespace:"", Metadata:(*model.NodeMetadata)(nil), SidecarScope:(*model.SidecarScope)(nil), MergedGateway:(*model.MergedGateway)(nil), ServiceInstances:[]*model.ServiceInstance(nil), IstioVersion:(*model.IstioVersion)(nil), ipv6Support:false, ipv4Support:false}
2020-03-16T02:48:08.352332Z	info	JWT policy is third-party-jwt`

### CODE REVIEW

**Comment by howardjohn** [comment]
/retest

There was a broken test that is fixed

**Comment by hzxuzhonghu** [comment]
TestFailover flakes?

File: pilot/cmd/pilot-agent/main.go [hunk 1]
```diff
@@ -163,6 +162,8 @@ var (
 				role.IPAddresses = append(role.IPAddresses, "::1")
 			}
 
+			role.IPAddresses = dedupeStrings(role.IPAddresses)
```
Comment (hzxuzhonghu) [inline pilot/cmd/pilot-agent/main.go]: I think the failure is related to the ip addrs orders changed

File: pilot/cmd/pilot-agent/main.go [hunk 2]
```diff
@@ -145,15 +145,23 @@ var (
...
 			} else if podIP != nil {
-				role.IPAddresses = append(role.IPAddresses, podIP.String())
+				role.IPAddresses = []string{podIP.String()}
 			}
 
 			// Obtain all the IPs from the node
 			if ipAddrs, ok := proxy.GetPrivateIPs(context.Background()); ok {
 				log.Infof("Obtained private IP %v", ipAddrs)
-				role.IPAddresses = append(role.IPAddresses, ipAddrs...)
+				if len(role.IPAddresses) == 1 {
```
Comment (ramaraochavali) [inline pilot/cmd/pilot-agent/main.go]: is deduping not helping this case?

Comment (hzxuzhonghu) [reply pilot/cmd/pilot-agent/main.go]: yes, the order matters. The podip must be the first one i think

Comment (ramaraochavali) [reply pilot/cmd/pilot-agent/main.go]: Ok. Makes sense. Then why don't need else block right? Any way by the time it comes here `len(role.IPAddresses)` will always be 1. Also can you please add comment about the importance of order here?

Comment (hzxuzhonghu) [reply pilot/cmd/pilot-agent/main.go]: I am not sure when it can be empty, you can see L168

Comment (ramaraochavali) [reply pilot/cmd/pilot-agent/main.go]: Yes. It can be empty or 1 - So We do not have to append incase it is empty right? Am I missing some thing?

Comment (ramaraochavali) [reply pilot/cmd/pilot-agent/main.go]: I see your point. Ignore the above comment

**Pull request merged**